[Docs] Setting `min_replicas` and `max_replicas` in `KafkaSource` (#5010)
mlrun/mlrun: docs/serving/graph-ha-cfg.md

lost. The mechanism of Window ACK provides a solution for such 
stateful event processing.

```{admonition} Note

> For statefeul functions, each worker has its own state.

Reviewer: Maybe mention `MapWithState` as an example of this.

Author: added

> For statefull functions, each worker has its own state. See, for example, {py:meth}`~storey.transformations.MapWithState`.

Reviewer: ```suggestion
For stateful functions, each worker has its own state. See, for example, {py:meth}`~storey.transformations.MapWithState`.
```

A consumer function is essentially a Nuclio function with a stream trigger. As part of the trigger, you can set a consumer group.  

> When the consumer function is part of a graph then the number of replicas per function depends on the  source:

Reviewer: I don't think this is specific to the function being "a part of a graph" (meaning a multi-function graph?). Rather, it's always the case that we applies this logic to control the number of replicas.

Author: updated

> - StreamSource: The number of replicas is derived from the number of shards and is therefore nonconfigurable. The same applies to the number of workers in each replica, which is set to 1 and is not configurable.  

Reviewer: I think this is clearer (because it being "the same" is initially ambiguous):
```suggestion
- `StreamSource`: The number of replicas is derived from the number of shards and is therefore nonconfigurable. Furthermore, the number of workers in each replica is set to 1 and is also not configurable.  
```

Author: updated
